llvm/include/llvm/ADT/ArrayRef.h

```diff
@@ -198,7 +198,10 @@ namespace llvm {
     }
 
     /// slice(n) - Chop off the first N elements of the array.
-    ArrayRef<T> slice(size_t N) const { return slice(N, size() - N); }
+    ArrayRef<T> slice(size_t N) const {
+      assert(N <= size() && "Invalid specifier");
+      return slice(N, size() - N);
```

Comment:
Instead, we could call into `drop_front` which does the same thing. Or make `drop_front` call slice.

Reply:
I think `drop_front` is more idiomatic and we could probably mark the single-argument slice as deprecated and eventually remove it. Although that's all out of scope for this PR. cc: @kazutakahirata

Reply:
Agree.